### DISCUSSION TITLE
How to implement i18n

### DISCUSSION THREAD

**Discussion by ThePromidius** [alejandroautalan/pygubu-designer#234] [Q&A] [answered]
So i've seen in pygubu-designer the option to add localization via i18n. However it's not referenced anywhere i could find. Tips or any guidance where to look to implement it?

**Comment by alejandroautalan** [comment, marked as answer]
Hello @ThePromidius thanks for trying pygubu.

To implement internationalization (I18N) I use the python [gettext](https://docs.python.org/3/library/gettext.html) module.

I made a simple example as a guide for implementation [here](https://github.com/alejandroautalan/pygubu-designer/tree/master/examples/i18n_gettext_demo).

Let me know if you have more questions.

Regards
Alejandro A.

**Comment by ThePromidius** [comment]
Hey! Thank you for going as far as making an example!
Just one last question. If i currently have a "This is a label" inside a label text field, should i make that a key in pygubu_designer (or something unique) like "widget_x_at_frame_x_label"?

I plan to make a weblate to let the community localize my program so thinking on best practices.

**Comment by ThePromidius** [comment]
Managed to get it working.

I guess that my question is self-answerable if i create a "en" lang and make it the default one or smth.
I'd be grateful if you can confirm it. Not a big deal to leave it as normal strings but what if they have special characters. Just making sure.

tried downloading gettext from gnu but was really outdated. ended up using the release from the poedit dev
https://github.com/vslavik/gettext-tools-windows/releases

Really thankfull for your support. (Btw i made a wiki entry with issues i found when building with pyinstaller in case you think it fits somewhere else)

**Reply by alejandroautalan** [reply]
Hello, sorry for the late response.

> Managed to get it working.
> 
> I guess that my question is self-answerable if i create a "en" lang and make it the default one or smth. I'd be grateful if you can confirm it.

Yes this is the most common way to do it.

> Really thankfull for your support. (Btw i made a wiki entry with issues i found when building with pyinstaller in case you think it fits somewhere else)

Thank you for your support. I will accommodate the it as soon as possible.

Regards
